### ISSUE TITLE
[AutoFill] 1Password cannot recognize TextField inputType on Android

### ISSUE THREAD

**Issue by AugustoResende** [flutter/flutter#105241]
## Use case

Autofill should work with password field, that currently not works and 1Password team says that there changes is needed from Flutter side. 

https://1password.community/discussion/127475/autofill-password-field-not-triggering-on-android-devices-for-flutter

> Thanks for letting us know that 1Password is not filling into the password field in your Flutter app on Android.
>
>After trying out a few different Flutter apps, I can confirm that the input type for the field is not shared with 1Password. This is an additional requirement that 1Password has for password fields where the input type must also be of a password type. Since we cannot verify the input type, 1Password will not recognize the field for filling the user’s password.
>
>We are excited to see Autofill support making its way into Flutter and we hope that we'll continue to see support for other Autofill features including sharing the input type in the near future.

>this feature needs to be supported in Flutter to match the native Android app experience

## Proposal

Maybe it's related to TYPE_TEXT_VARIATION_PASSWORD android input types?

**Comment by maheshmnj** [comment]
Hi @AugustoResende, Thanks for filing the issue. Could you please help me with the reproducible steps along with the output of `flutter doctor -v` to investigate further?  

Thanks

**Comment by dupuchba** [comment]
It has been identified by the 1Password team -> https://1password.community/discussion/127475/autofill-password-field-not-triggering-on-android-devices-for-flutter
It's not working for *all* version of Flutter apparently.
Citing the 1Password engineer
>After trying out a few different Flutter apps, I can confirm that the input type for the field is not shared with 1Password. This is an additional requirement that 1Password has for password fields where the input type must also be of a password type. Since we cannot verify the input type, 1Password will not recognize the field for filling the user’s password.

Basically @AugustoResende solution would seem fine but I am pretty sure that `TextField` do not end up being converted to platform code.

**Comment by AugustoResende** [comment]
@maheshmnj i don't think that flutter doctor -v is needed

Steps:
Create a flutter app with autofillgroup password
Install 1Password
Create a login/password for your app in 1Password
Make 1Password your default autofill in your android device (emulator too)
Go to your flutter app
Do autofill
Only e-mail is filled, password still empty.

**Comment by maheshmnj** [comment]
Thanks for the link @dupuchba, I am labeling this issue as a bug based on the [response from 1Password team member](https://1password.community/discussion/comment/631456/#Comment_631456)

> After trying out a few different Flutter apps, I can confirm that the input type for the field is not shared with 1Password. This is an additional requirement that 1Password has for password fields where the input type must also be of a password type. Since we cannot verify the input type, 1Password will not recognize the field for filling the user’s password.

**Comment by EthanIsaac** [comment]
Hello! Any updates on this? I'm trying to implement autofill for my flutter app but still not working.

**Comment by dupuchba** [comment]
@EthanIsaac Like explained earlier, `TextField` are not platform code and 1Password does not know how to interpret them.
I think best chance for this would be to ask 1Password team if there are other ways to make 1Password understand that our TextField can be autocompleted

**Comment by ricardoboss** [comment]
Over on the 1password forum, they are saying you need to implement an Android native compatible way to provide the auto-fillable information... Can one of you decide who has to implement it? Can we as the users do anything to help implement it on either side?

**Comment by dupuchba** [comment]
@ricardoboss like I wrote above, `TextField`s are not native component (fortunately). If the `1Password` team does not adapt its way of identifying a password input I don't see what we can do.

**Comment by rekire** [comment]
What's the current status here?

**Comment by ricardoboss** [comment]
@dupuchba I can use Google with the same example app (https://github.com/ricardoboss/flutter_password_autofill) and Google is able to fill in my credentials. Maybe someone who worked on this implementation might be able to help the 1password devs?

**Comment by nealsanche** [comment]
I looked deeply into this today. I got one of the Google auto fill service examples running, and checked what the code sees when a password field is focused in a Flutter app I'm working on. In the sample code it iterates through the view nodes that flutter is producing, and it is able to properly get the autofill hints, however asking what the `inputType` of the node is always returns 0. The AutoFillType is TEXT which seems fine. Here's a little output from the run:

```
D/BasicService: inputType = 0
D/BasicService: autofillType = 1
D/BasicService: autoFillHints = [password]
V/BasicService: Setting hint 'password' on 1073741824:543304862@2145823823
D/BasicService: autofillable fields:{password=1073741824:543304862@2145823823}
```

I believe 1Password is saying that they also require `inputType` to be set to Password. I've asked if it would be possible to relax this requirement as a user setting in the 1Password app. Really, this stuff should be that hard, but I suppose, being a security conscious company, they want to make sure the field input type and the autofill hints are exactly right.

I guess my question here is can Flutter provide the correct `inputType` for a field that's not native?
